tests/failpoint/failpoint.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package failpoint
+
+import (
+	gofail "go.etcd.io/gofail/runtime"
+)
+
+type Failpoint interface {
+	Enable() error
+	Disable() error
+}
+
+type failpointFunc struct {
+	name    string
+	payload string
+}
+
+func (f *failpointFunc) Enable() error {
+	return gofail.Enable(f.name, f.payload)
```

Comment:
It isn't enough, you still need to run command `gofail enable <package> ` to transform the source in advance.

Reference:
- https://github.com/etcd-io/bbolt/blob/8b1ee10512ccb01d9d8744a620afc12939d26fb2/Makefile#L65-L66
- https://github.com/etcd-io/etcd/blob/b404d25d842cff4a281d1debdff764b2b126e74e/tests/robustness/makefile.mk#L38

Reply:
Yeah, I intentionally leave it in line 22 of `tests/failpoint/cluster_test.go` as a comment. I wanted to enable this test verification in the CI one the fix is merged. So this test can only be run manually in local box.